plugins/modules/dcnm_service_node.py

```diff
@@ -0,0 +1,741 @@
+    state: merged 
+    config:
+    - name: SN-11
+      type: firewall
+      form_factor: virtual
+      svc_int_name: svc1
+      attach_interface: Ethernet1/1
+      switches:
+      - 192.168.1.224
+    - name: SN-12
+      type: firewall
+      form_factor: virtual
+      svc_int_name: svc1
+      attach_interface: vPC1
+      switches:  # up to two switches, if two switches are provided, vpc is only option
+      - 192.168.1.224
+      - 192.168.1.225
+
+- name: Replace Service Nodes form factor/type parameter
+  cisco.dcnm.dcnm_service_node:
+    fabric: Fabric1
+    service_fabric: external
+    state: replaced
+    config:
+    - name: SN-11
+      type: firewall
+   #  Replace can only modify the form factor
+   #  form_factor: virtual  # the virtual will be changed to new physical
+   #  form_factor: physical
+      svc_int_name: svc1
+      attach_interface: Ethernet1/1
+      switches:
+      - 192.168.1.224
+   #   Nothing will be replaced in the below service node as there is no change 
+   #   Dont touch this if its present on DCNM
+   # - name: SN-12
+   #   type: firewall
+   #   form_factor: virtual
+   #   svc_int_name: svc1
+   #   attach_interface: vPC1
+   #   switches:  # up to two switches, if two switches are provided, vpc is only option
+   #   - 192.168.1.224
+   #   - 192.168.1.225
+
+- name: Override Service Nodes
+  cisco.dcnm.dcnm_service_node:
+    fabric: Fabric1
+    service_fabric: external
+    state: overridden
+    config:
+   # Create this service node
+     - name: SN-13
+      type: firewall
+      form_factor: virtual
+      svc_int_name: svc1
+      attach_interface: Ethernet1/1
+      switches:
+      - 192.168.1.224
+   # Delete this service node from the DCNM
+   # - name: SN-11
+   #   type: firewall
+   #   form_factor: virtual
+   #   svc_int_name: svc1
+   #   attach_interface: Ethernet1/1
+   #   switches:
+   #   - 192.168.1.224 
+   # Delete this service node from the DCNM
+   # - name: SN-12
+   #   type: firewall
+   #   form_factor: virtual
+   #   svc_int_name: svc1
+   #   attach_interface: vPC1
+   #   switches:  # up to two switches, if two switches are provided, vpc is only option
+   #   - 192.168.1.224
+   #   - 192.168.1.225
+
+- name: Delete selected Service Nodes
+   cisco.dcnm.dcnm_service_node:
+    fabric: Fabric1
+    service_fabric: external
+    state: deleted
+    config:
+    - name: SN-11
+      type: firewall
+      form_factor: virtual
+      svc_int_name: svc1
+      attach_interface: Ethernet1/1
+      switches:
+      - 192.168.1.224
+    - name: SN-12
+      type: firewall
+      form_factor: virtual
+      svc_int_name: svc1
+      attach_interface: vPC1
+      switches:  # up to two switches, if two switches are provided, vpc is only option
+      - 192.168.1.224
+      - 192.168.1.225
+
+- name: Delete all the Service Nodes
+  cisco.dcnm.dcnm_service_node:
+    fabric: Fabric1
+    service_fabric: external
+    state: deleted
+
+- name: Query Service Nodes
```

Comment:
Just to clarify can you update to state `Query Service Node state for SN-11 and SN-12`

Reply:
Addressed.